app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
Do you have a tip regarding implementation?
On Jul 24, 2015 8:04 PM, "Arnold Sykosch" <arnold.sykosch@gmail.com> wrote:

> That is a good point. I should be noted on the modal with the file
> chooser. And yes, I totally agree on the seamless integration.
> On Jul 24, 2015 7:58 PM, "Stan Hu" <notifications@github.com> wrote:
>
> In app/controllers/projects/blob_controller.rb
> <https://github.com/gitlabhq/gitlabhq/pull/9138#discussion_r35448408>:
>
> >          File.join(@path, File.basename(params[:file_name]))
> >        else
> >          @path
> >        end
> > +
> > +    if params[:file_upload].present?
> > +      params[:content] = params[:file_upload].read
> > +      params[:encoding] = 'text'
>
> Perhaps, but then this feature needs to work seamlessly for both text and
> binary files.
>
> We also have to consider what happens when the file uploaded is too big.
> Dropzone used in GitLab prevents users from sending anything over 10 MB,
> and the Web server also has to be configured appropriately to accept
> payloads that large.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gitlabhq/gitlabhq/pull/9138/files#r35448408>.
>
>